druid/src/core.rs

```diff
@@ -765,6 +882,68 @@ impl<'a, 'b> EventCtx<'a, 'b> {
     pub fn widget_id(&self) -> WidgetId {
         self.widget_id
     }
+
+    pub(crate) fn make_lifecycle_ctx(&mut self) -> LifeCycleCtx {
+        LifeCycleCtx {
+            command_queue: self.command_queue,
+            children_changed: false,
+            needs_inval: false,
+            children: Bloom::default(),
+            request_anim: false,
+            window_id: self.window_id,
+            widget_id: self.widget_id,
+        }
+    }
+}
+
+impl<'a> LifeCycleCtx<'a> {
+    /// Invalidate.
+    ///
+    /// See [`EventCtx::invalidate`](struct.EventCtx.html#method.invalidate) for
+    /// more discussion.
+    pub fn invalidate(&mut self) {
+        self.needs_inval = true;
+    }
+
+    /// Returns the current widget's `WidgetId`.
+    pub fn widget_id(&self) -> WidgetId {
+        self.widget_id
+    }
+
+    /// Registers a child wildget. This should only be called
```

Comment:
In a list view where a child is added dynamically in response to a data change, does the widget call `children_changed` to request a `RegisterChildren` event? If so, might be worth doing a doc link here. Also typo "widget".

Reply:
that is the current flow, yes. This gets improved docs in a future patch; an important detail is that the only responsibility of a container widget is to ensure that this event reaches each child `WidgetPod`; the pod handles the actual registering of the child, and you shouldn't have to call this directly unless you're doing something very fancy.

I've also been thinking that we may be able to unify `WidgetAdded` and `RegisterChildren`, which would be a nice simplification; ideally there is a single place where the widget does various one-time tasks; another of these tasks (coming up soon) is registering to participate in tab-to-focus.